addons/sale_coupon/models/sale_order.py

```diff
@@ -266,12 +266,11 @@ def _get_reward_values_percentage_amount(self, program):
 
                 if discount_line_amount:
 
-                    if line.tax_id in reward_dict:
-                        reward_dict[line.tax_id]['price_unit'] -= discount_line_amount
+                    taxes = self.fiscal_position_id.map_tax(line.tax_id).filtered(lambda t: t.amount_type != 'fixed')
+                    if taxes in reward_dict:
+                        reward_dict[taxes]['price_unit'] -= discount_line_amount
                     else:
-                        taxes = self.fiscal_position_id.map_tax(line.tax_id).filtered(lambda t: t.amount_type != 'fixed')
-
-                        reward_dict[line.tax_id] = {
```

Comment:
This is the only part of your changes that is tested, the two others are not :eyes: (the test is green even if I revert the two other changes from `line.tax_id` to `taxes`)

I won't ask you to rewrite an additional test, it's ok :D

Reply:
Thanks ! :sweat_smile: